rich/highlighter.py

```diff
@@ -94,6 +94,7 @@ class ReprHighlighter(RegexHighlighter):
             r"(?P<call>[\w.]*?)\(",
             r"\b(?P<bool_true>True)\b|\b(?P<bool_false>False)\b|\b(?P<none>None)\b",
             r"(?P<ellipsis>\.\.\.)",
+            r"(?P<complex_number>(?<!\w)(?:\-?[0-9]+\.?[0-9]*(?:e[-+]?\d+?)?|0x[0-9a-fA-F]*)(?:[-+](?:[0-9]+\.?[0-9]*(?:e[-+]?\d+)?|0x[0-9a-fA-F]*))?j)",
```

Comment:
We might be able to simplify the regex. What we're formatting is the output of repr strings, which doesn't produce hex (that I'm aware of), so I think we can drop the `0x...` part.

Reply:
done! (in 69003a772c)